File: 1201-1300/1299-Replace-Elements/1299.py
class Solution:
  def replaceElements(self, arr: List[int]) -> List[int]:
    n = len(arr)
    ans = [0] * (n - 1) + [-1]
    for i in range(n - 2, -1, -1):
      ans[i] = max(ans[i + 1], arr[i + 1])
    return ans

File: 1201-1300/1299-Replace-Elements/1299.java
class Solution {
  public int[] replaceElements(int[] arr) {
    int rightMax = arr[arr.length - 1];
    arr[arr.length - 1] = -1;
    for (int i = arr.length - 2; i >= 0; i--) {
      int t = arr[i];
      arr[i] = rightMax;
      if (t > rightMax)
        rightMax = t
    }
    return arr;
  }
}